### PR TITLE
rustdoc: render visibility on associated types

### PR DESCRIPTION
**Pull request by wackbyte**
This should only affect inherent associated types (#8995).

### CODE REVIEW

File: tests/rustdoc/anchors.no_type_anchor2.html [hunk 1]
```diff
@@ -1 +1 @@
-<section id="associatedtype.Y" class="associatedtype"><h4 class="code-header">type <a href="#associatedtype.Y" class="associatedtype">Y</a> = <a class="primitive" href="{{channel}}/std/primitive.u32.html">u32</a></h4></section>
\ No newline at end of file
+<section id="associatedtype.Y" class="associatedtype"><h4 class="code-header">pub type <a href="#associatedtype.Y" class="associatedtype">Y</a> = <a class="primitive" href="https://doc.rust-lang.org/nightly/std/primitive.u32.html">u32</a></h4></section>
```
Comment (fmease) [inline tests/rustdoc/anchors.no_type_anchor2.html:1]: Please keep the channel placeholder

```suggestion
<section id="associatedtype.Y" class="associatedtype"><h4 class="code-header">pub type <a href="#associatedtype.Y" class="associatedtype">Y</a> = <a class="primitive" href="{{channel}}/std/primitive.u32.html">u32</a></h4></section>
```

**Review by fmease — CHANGES_REQUESTED** [review]
Thanks for catching and fixing this! I've marked one thing that definitely needs to be addressed.
Apart from that, I think it would be good to add at least one more test (file) that explicitly checks for the new behavior (e.g. with `@has`) instead of relying on this `@snapshot` test that could accidentally get `--bless`ed in the future without notice.

**Comment by wackbyte** [comment]
I added the test to the existing `visibility` test with tests for other associated items as well.

**Review by fmease — APPROVED** [review]
Thanks for adding the test :+1:

LGTM (I personally don't have bors rights, so I cannot bors-approve)

**Comment by jsha** [comment]
Thanks for the fix, @wackbyte! Sorry I took so long to review this. Looks good to me as well - and thanks @fmease for the review.

@bors r+ rollup

**Pull request merged**